src/Compiler/Opts/InlineHeuristics.idr

```diff
@@ -55,7 +56,7 @@ inlineHeuristics fn = do
     -- intervention by this function.
     -- We could lift this public restriction if we checked that the source def was _either
     -- public OR the destination was the same module as the source_.
-    let Public = gdef.visibility
+    let Value Public = gdef.visibility
```

Comment:
I don't think you should ever explicitly match on `Value` as that breaks the abstraction
that we should default to the default value.

I think it makes sense to have `WithDefault`  just `export`ed (rather than `public export`ed)
and a smart constructor `value` defined to prevent these kind of matches.

Reply:
Makes sense, yeah. I'll add `value : a -> WithDefault a def` and `defaultValue : WithDefault a def`.